src/main/java/it/gov/pagopa/pu/debtpositions/service/statusalign/DebtPositionHierarchyStatusAlignerServiceImpl.java

```diff
@@ -0,0 +1,77 @@
+import it.gov.pagopa.pu.debtpositions.dto.generated.IupdSyncStatusUpdateDTO;
+import it.gov.pagopa.pu.debtpositions.mapper.DebtPositionMapper;
+import it.gov.pagopa.pu.debtpositions.model.DebtPosition;
+import it.gov.pagopa.pu.debtpositions.repository.DebtPositionRepository;
+import it.gov.pagopa.pu.debtpositions.repository.InstallmentNoPIIRepository;
+import it.gov.pagopa.pu.debtpositions.service.statusalign.debtposition.DebtPositionInnerStatusAlignerService;
+import it.gov.pagopa.pu.debtpositions.service.statusalign.paymentoption.PaymentOptionInnerStatusAlignerService;
+import jakarta.transaction.Transactional;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.Map;
+
+import static it.gov.pagopa.pu.debtpositions.dto.generated.InstallmentStatus.TO_SYNC;
+
+@Service
+@Slf4j
+public class DebtPositionHierarchyStatusAlignerServiceImpl implements DebtPositionHierarchyStatusAlignerService {
+
+  private final DebtPositionRepository debtPositionRepository;
+  private final InstallmentNoPIIRepository installmentNoPIIRepository;
+  private final PaymentOptionInnerStatusAlignerService paymentOptionInnerStatusAlignerService;
+  private final DebtPositionInnerStatusAlignerService debtPositionInnerStatusAlignerService;
+  private final DebtPositionMapper debtPositionMapper;
+
+
+  public DebtPositionHierarchyStatusAlignerServiceImpl(DebtPositionRepository debtPositionRepository,
+                                                       InstallmentNoPIIRepository installmentNoPIIRepository, PaymentOptionInnerStatusAlignerService paymentOptionInnerStatusAlignerService, DebtPositionInnerStatusAlignerService debtPositionInnerStatusAlignerService, DebtPositionMapper debtPositionMapper) {
+    this.debtPositionRepository = debtPositionRepository;
+    this.installmentNoPIIRepository = installmentNoPIIRepository;
+    this.paymentOptionInnerStatusAlignerService = paymentOptionInnerStatusAlignerService;
+    this.debtPositionInnerStatusAlignerService = debtPositionInnerStatusAlignerService;
+    this.debtPositionMapper = debtPositionMapper;
+  }
+
+  @Transactional
+  @Override
+  public DebtPositionDTO finalizeSyncStatus(Long debtPositionId, Map<String, IupdSyncStatusUpdateDTO> syncStatusDTO) {
+    DebtPosition debtPosition = debtPositionRepository.findOneWithAllDataByDebtPositionId(debtPositionId);
+
+    debtPosition.getPaymentOptions().forEach(paymentOption ->
+      paymentOption.getInstallments().stream()
+        .filter(installment -> {
+          boolean isToSync = TO_SYNC.equals(installment.getStatus());
+          boolean hasIud = syncStatusDTO.containsKey(installment.getIud());
+
+          if (!hasIud) {
+            log.error("Installment with IUD [{}] is not present in the syncStatusDTO map", installment.getIud());
+          } else if (!isToSync) {
+            log.error("Installment with IUD [{}] does not have TO_SYNC status", installment.getIud());
+          }
```

Comment:
these conditions are wrong, the errors are:
1. iud2update & !isToSync
2. !iud2Update & isToSync
Thus, the error condition to log is if the booleans have different values!